packages/babel-parser/src/plugins/flow.js

```diff
@@ -2525,10 +2527,12 @@ export default (superClass: Class<Parser>): Class<Parser> =>
         }
       } else if (
         specifierTypeKind !== null &&
-        (this.match(tt.name) || this.state.type.keyword)
+        (this.match(tt.name) ||
+          this.match(tt.string) ||
+          this.state.type.keyword)
       ) {
         // `import {type foo`
-        specifier.imported = this.parseIdentifier(true);
+        specifier.imported = this.parseModuleExportName();
```

Comment:
I don't think `import { type "foo" } from "bar"` is supported in flow, neither implied from `moduleStringNames`. 
Since `moduleStringNames` are meant to offer compatibility for WASM modules, I don't think Flow will practically support it in type imports.